pkg/kubelet/status/status_manager.go

```diff
@@ -414,7 +414,8 @@ func (m *manager) syncBatch() {
 
 // syncPod syncs the given status with the API server. The caller must not hold the lock.
 func (m *manager) syncPod(uid types.UID, status versionedPodStatus) {
-	if !m.needsUpdate(uid, status) {
+	// always allow status updates if the pod can be deleted.
+	if !m.needsUpdate(uid, status) && !m.couldBeDeleted(uid, status.status) {
```

Comment:
This is still fragile since you have to keep the code at two places in sync. Maybe add another method dedicated for the check?